src/MLIR/AST/Builder.hs

```diff
@@ -134,6 +140,7 @@ instance MonadNameSupply m => MonadBlockBuilder (BlockBuilderT m) where
     value <- lift $ freshValue ty
     modify \s -> s { blockArguments = blockArguments s .:. value }
     return value
+  setLocation loc = BlockBuilderT $ modify \s -> s { blockLocation = loc }
```

Comment:
Given that we don't override known locations, perhaps we should rename this to `setDefaultLocation` (and the `blockLocation` -> `blockDefaultLocation`)?